aten/src/TH/generic/THTensorRandom.cpp

```diff
@@ -247,9 +247,9 @@ void THTensor_(multinomialAliasDraw)(THLongTensor *self, THGenerator *_generator
 
       J_sample = THLongTensor_fastGet1d(J, rand_ind);
 
-      sample_idx = J_sample*(1 -_mask) + (rand_ind+1L) * _mask;
+      sample_idx = J_sample*(1 -_mask) + rand_ind * _mask;
```

Comment:
Just to confirm - we don't need this fix for the CUDA implementation?

Reply:
There isn’t a CUDA implementation, unfortunately.

Reply:
I see the multinomial alias draw and setup methods in `THTensorRandom.cu`. I'm guessing these tests should exercise that on CUDA?

Reply:
Just looked at the test failures, I think the CUDA test failures are related.

Reply:
Oh I’m sorry. My memory failed me. Yes, the CUDA implementations are also available and are tested. I’ll investigate the issue.